repokid/cli/reactor_cli.py

```diff
@@ -0,0 +1,192 @@
+        return (True, return_val)
+
+
+def opt_out(dynamo_table, message, config=None):
+    if not message.reason or not message.requestor:
+        return (False, 'Reason and requestor must be specified')
+
+    role_id = dynamo.find_role_in_cache(dynamo_table, message.account, message.role_name)
+
+    if not role_id:
+        return (False, 'Unable to find role {} in account {}'.format(message.role_name, message.account))
+
+    role_data = dynamo.get_role_data(dynamo_table, role_id, fields=['OptOut'])
+    if 'OptOut' in role_data and role_data['OptOut']:
+
+        return (False, 'Role {} in account {} is already opted out by {} for reason {} until {}'.format(
+            message.role_name, message.account, role_data['OptOut']['owner'], role_data['OptOut']['reason'],
+            time.strftime('%m/%d/%y', time.localtime(role_data['OptOut']['expire']))))
+
+    else:
+        current_dt = datetime.datetime.fromtimestamp(time.time())
+        expire_dt = current_dt + datetime.timedelta(config['opt_out_period_days'])
+        expire_epoch = int((expire_dt - datetime.datetime(1970, 1, 1)).total_seconds())
+        new_opt_out = {'owner': message.requestor, 'reason': message.reason, 'expire': expire_epoch}
+        dynamo.set_role_data(dynamo_table, role_id, {'OptOut': new_opt_out})
+        return (True, 'Role {} in account {} opted-out until {}'.format(message.role_name, message.account,
+                                                                        expire_dt.strftime('%m/%d/%y')))
+
+
+def remove_opt_out(dynamo_table, message, config=None):
+    role_id = dynamo.find_role_in_cache(dynamo_table, message.account, message.role_name)
+
+    if not role_id:
+        return (False, 'Unable to find role {} in account {}'.format(message.role_name, message.account))
+
+    role_data = dynamo.get_role_data(dynamo_table, role_id, fields=['OptOut'])
+
+    if 'OptOut' not in role_data or not role_data['OptOut']:
+        return (False, 'Role {} in account {} wasn\'t opted out'.format(message.role_name, message.account))
+    else:
+        dynamo.set_role_data(dynamo_table, role_id, {'OptOut': {}})
+        return (True, 'Cancelled opt-out for role {} in account {}'.format(message.role_name, message.account))
+
+
+def rollback_role(dynamo_table, message, config=None):
+    if not message.selection:
+        return (False, 'Rollback must contain a selection number')
+
+    errors = cli.rollback_role(message.account, message.role_name, dynamo_table, CONFIG, selection=message.selection,
+                               commit=True)
+    if errors:
+        return (False, 'Errors during rollback: {}'.foramt(errors))
+    else:
+        return (True, 'Successfully rolled back role {} in account {}'.format(message.role_name, message.account))
+
+
+def send_message(sns_object, topic_arn, message_dict):
+    sns_object.publish(TopicArn=topic_arn, Message=json.dumps(message_dict))
```

Comment:
I would dump `init_messaging(...)` and put the `@sts_conn` decorator on `send_message`:

```python
from cloudaux.aws.sts import sts_conn

@sts_conn('sns')
def send_message(topic_arn, message_dict, client=None):
    client.publish(TopicARN=topic_arn, Message=json.dumps(message_dict)
```